src/services/BackendService/BackendService.ts

```diff
@@ -92,10 +102,47 @@ export class BackendService implements BackendInterface {
     };
   }
 
+  filterTEKs = (contagiousDateInfo?: ContagiousDateInfo) => {
+    return (key: TemporaryExposureKey) => {
+      // This function filters out TEKs that were generated before the user was contagious.
+      // rollingStartIntervalNumber = A number describing when a key starts. It is equal to
+      // startTimeOfKeySinceEpochInSecs / (60 * 10).
+      // rollingPeriod = A number describing how long a key is valid. It is expressed in
+      // increments of 10 minutes (e.g. 144 for 24 hours).
+      // source: https://developers.google.com/android/reference/com/google/android/gms/nearby/exposurenotification/TemporaryExposureKey
+      if (!contagiousDateInfo || contagiousDateInfo.dateType === 'noDate' || !contagiousDateInfo.dateString) {
+        return true;
+      }
+      // the following is a bit hacky - maybe we should pass the date as a Date, not a string
+      const dateParts = contagiousDateInfo.dateString.split('-');
+      const providedDate = new Date(Number(dateParts[0]), Number(dateParts[1]) - 1, Number(dateParts[2]));
+      const providedDateHoursSinceEpoch = hoursSinceEpoch(providedDate);
+      let contagiousStartHoursSinceEpoch;
+      if (contagiousDateInfo.dateType === 'symptomOnsetDate') {
+        contagiousStartHoursSinceEpoch = providedDateHoursSinceEpoch - CONTAGIOUS_DAYS_BEFORE_SYMPTOM_ONSET * 24;
+      } else {
+        contagiousStartHoursSinceEpoch = providedDateHoursSinceEpoch - CONTAGIOUS_DAYS_BEFORE_TEST_DATE * 24;
+      }
+
+      const rollingEndIntervalNumber = key.rollingStartIntervalNumber + key.rollingPeriod;
+      const rollingEndIntervalHoursSinceEpoch = rollingEndIntervalNumber / 6;
```

Comment:
What is 6?

Reply:
`rollingEndIntervalNumber` is the number of 10 minute increments since epoch (see my comment [here](https://github.com/cds-snc/covid-alert-app/pull/1094/files/f9aae64e7df2e670487b3e6c6147231730878c57#diff-4faa9e63d84083d73a7e13a538ff797eR110)). So I am dividing by 6 to get that in hours.

Reply:
There should be a comment in the code to indicate that, or if the 6 is used somewhere else, then defined in a variable. Otherwise it's just a magic number.